### PR TITLE
Horovod: adjust base LR used by schedulers to scale with the number of workers

### PR DESCRIPTION
**Pull request by tgaddair**
In #2574 it was observed that the learning rate used to initialize learning rate schedulers is in conflict with the way we scale the learning rate with the number of Horovod workers.  Because the learning rate schedulers are initialized before the scaling, the scaling would be overridden.  This PR fixes this so that optimizers and LR schedulers scale up correctly.

Follow-ups to consider would be:

- Make scaling work for other backends like DDP.
- Add an option to configure the LR scaling in the Trainer.

### CODE REVIEW

**Comment by tgaddair** [comment]
@williamFalcon @Borda failing tests seem unrelated.  Could you take a look at the PR and see if everything makes sense?

As a follow-up, I want to add a param to `Trainer` so we can override the LR scale when using Horovod/DDP.

**Comment by Borda** [comment]
> @williamFalcon @Borda failing tests seem unrelated. Could you take a look at the PR and see if everything makes sense?

yes, no connection to this PR

File: tests/models/test_horovod.py [hunk 1]
```diff
@@ -175,3 +177,36 @@ def get_optimizer_params(optimizer):
...
+            limit_train_batches=0.2,
+            distributed_backend='horovod'
+        )
+        results = trainer.fit(model)
+        assert results == 1
+
+    adjusted_lr1 = [pg['lr'] for pg in trainer.optimizers[0].param_groups][0]
+    adjusted_lr2 = [pg['lr'] for pg in trainer.optimizers[1].param_groups][0]
+
+    # Called ones after end of epoch
```
Comment (Borda) [inline tests/models/test_horovod.py:208]:
```suggestion
    # Called ones after end of epoch with gamma=0.1
```

File: tests/models/test_horovod.py [hunk 2]
```diff
@@ -175,3 +177,36 @@ def get_optimizer_params(optimizer):
...
+    assert pytest.approx(init_lr * 0.1) == adjusted_lr1
+
+    # Called every 3 steps, meaning for 1 epoch of 11 batches, it is called 3 times
```
Comment (Borda) [inline tests/models/test_horovod.py:211]:
```suggestion
    # Called every 3 steps, meaning for 1 epoch of 11 batches, it is called 3 times with gamma=0.1
```

**Comment by amorehead** [comment]
Hello. I was curious if there have been any other PRs that have added the ability for users to specify which learning rate scaling strategy should be used with e.g., DDP/Horovod.

**Pull request merged**